libs/multi-chain/src/lib/near/signer.ts

```diff
@@ -42,8 +53,18 @@ export class NearSigner extends Signer<NearSendTXOpts> {
     return this.initProm;
   }
 
-  public static getImplicitAccountId(pubkey: string) {
-    return Buffer.from(utils.PublicKey.fromString(pubkey).data).toString('hex');
+  private buildNativeAction(action: NearAction): NearNativeAction {
+    switch (action.type) {
+      case NearSupportedActionTypes.TRANSFER:
+        if (action.params.discriminator !== NearSupportedActionTypes.TRANSFER) {
+          throw 'the input parameters do not match the call';
+        }
+        return transactions.transfer(
+          new BN((action.params as NearTransferParam).amount)
+        );
+      default:
+        throw `Action of type ${action.type} is unsupported`;
+    }
   }
 
   public async sendTX(opts: NearSendTXOpts) {
```

Comment:
can we separate this into `createTx`, `signTx` and `sendTx` such that you can use one without the othesr, ideally making them default static but overload with a non-static method? (can you tell why I don't like OOP yet :P)

For instance, creating the transaction  `sendTx` in theory only requires a pubkey and networkId and I could imagine many cases where someone would want to create a tx without signing it (pretty much any Dapp). Another example is cases where someone would want to sign a tx without sending it directly to the near blockchain - for instance, maybe when using some layer 2.

Reply:
I seperated out into different functions. I think adding a static overload for now might be over specing. Like, as long as its an easy thing to add later, we can add it when needed?

Reply:
Sure, just keep it in mind and do it before it's too late.